client/my-sites/themes/controller.jsx

```diff
@@ -32,6 +32,7 @@ export function getProps( context ) {
 		isCollectionView: view === 'collection',
 		pathName: context.pathname,
 		trackScrollPage: boundTrackScrollPage,
+		isServerSide: context.isServerSide,
```

Comment:
This seems to be causing the logged-out version to be displayed first as reported in https://github.com/Automattic/wp-calypso/pull/103080#pullrequestreview-2823263102. Since getProps() is also used in controller-logged-in:

https://github.com/Automattic/wp-calypso/blob/57332811d46ade401a0e52511acba7e76fbf440c/client/my-sites/themes/controller-logged-in.jsx#L24

Try adding `isServerSide` in the `loggedOut` function so that it's only applied for the LoggedOutComponent.

https://github.com/Automattic/wp-calypso/pull/103080/files#diff-610ef48de8034f07a32053763f52996c70f427622e3baeecc6b6ba3fb7d0574fR48

Reply:
I've completely removed the `isServerSide` property and tried another approach, wrapping the CustomSelectControl element and preventing SSR for components that aren't SSR-compatible. The approach works fine, but the issue with flickering between logged-in and logged-out state remains.

It seems that some other changes caused it. The problem is that it is not reproducible locally.